meson.build

```diff
@@ -142,6 +142,20 @@ add_project_arguments(
     language: 'c'
 )
 
+tcmalloc_variations = []
+if get_option('default_library') != 'shared'
+    if get_option('debug')
+        tcmalloc_variations += 'libtcmalloc_minimal_debug'
+    endif
+    tcmalloc_variations += 'libtcmalloc_minimal'
+else
+    if get_option('debug')
+        tcmalloc_variations += 'libtcmalloc_debug'
+    endif
+endif
+# Fallbacks in the event the variations selected are not available
+tcmalloc_variations += 'libtcmalloc'
+tcmalloc_dep = dependency(tcmalloc_variations, required: get_option('tcmalloc'))
```

Comment:
It's not clear to me how this would work.  If the library is compiled to use tcmalloc, but some app wants to use jemalloc, who wins?